File: Homework/themoviedb/src/main/java/com/example/themoviedb/Injection.java
package com.example.themoviedb;

import android.content.Context;

import com.example.themoviedb.persistence.LocalMovieDataSource;
import com.example.themoviedb.persistence.MoviesDatabase;

public class Injection {

    public static MovieDataSource provideMovieDataSource(Context context) {
        MoviesDatabase database = MoviesDatabase.getInstance(context);
        return new LocalMovieDataSource(database.movieDao());
    }
}

File: Homework/themoviedb/src/main/java/com/example/themoviedb/Injection.kt
package com.example.themoviedb

import android.content.Context

import com.example.themoviedb.persistence.LocalMovieDataSource
import com.example.themoviedb.persistence.MoviesDatabase

object Injection {

    fun provideMovieDataSource(context: Context?): MovieDataSource {
        val database = MoviesDatabase.getInstance(context)
        return LocalMovieDataSource(database!!.movieDao())
    }
}